synapse/storage/events.py

```diff
@@ -428,6 +428,7 @@ def _calculate_state_delta(self, room_id, events_context, new_latest_event_ids):
         # Now we need to work out the different state sets for
         # each state extremities
         state_sets = []
+        state_groups = set()
```

Comment:
I think its probably fine for integers, and I don't think this function will be hit enough for it to be a problem